### PR TITLE
done Iterator

### PR DESCRIPTION
**Pull request by IIioneR**

### CODE REVIEW

File: Iterator.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            self.start = args[0]
+            self.end = args[1]
+            self.step = 1.0
+            self.counter = self.start
+
+        elif len(args) == 3:
+            self.start = args[0]
+            self.end = args[1]
+            self.step = args[2]
+            self.counter = self.start
```
Comment (dbradul) [inline Iterator.py]: Consider simplifying it:
```
def __init__(self, arg0, arg1=None, step=1.0):
    if arg1 is not None:
        start, end = arg0, arg1
    else
        start, end = 0, arg0
    ...
```

File: Iterator.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            self.counter += self.step
+            return self.counter
+        else:
+            raise StopIteration
+
+
+frange = RangeFloatIterator(3, 50, 0.7)
+
+for i in frange:
+    print(i)
```
Comment (dbradul) [inline Iterator.py:35]: Doesn't pass the simple test:
```
assert(list(RangeFloatIterator(5)) == [0, 1, 2, 3, 4])
```

Comment (dbradul) [reply Iterator.py:35]: Consider creating tests similar to the mentioned above to cover base and corner cases.

**Review by dbradul — CHANGES_REQUESTED** [review]

File: Iterator.py [hunk 3]
```diff
@@ -4,33 +4,29 @@ class RangeFloatIterator:
...
 
 
-frange = RangeFloatIterator(3, 50, 0.7)
+frange = RangeFloatIterator(5, 10, 1.5)
 
 for i in frange:
     print(i)
+
+assert(list(RangeFloatIterator(5)) == [0.0, 1.0, 2.0, 3.0, 4.0])
+
```
Comment (dbradul) [inline Iterator.py]: This doesn't pass either:
```
assert(list(RangeFloatIterator(2, 5.5, 1.5)) == [2, 3.5, 5])
```

**Review by dbradul — CHANGES_REQUESTED** [review]

File: Iterator.py [hunk 4]
```diff
@@ -16,17 +16,19 @@ def __init__(self, start, end=None, step=1.0):
...
-    print(i)
-
 assert(list(RangeFloatIterator(5)) == [0.0, 1.0, 2.0, 3.0, 4.0])
+assert(list(RangeFloatIterator(5)) == [0, 1, 2, 3, 4])
+assert(list(RangeFloatIterator(2, 5.5, 1.5)) == [2.0, 3.5, 5.0])
+assert(list(RangeFloatIterator(0, 2, 1)) == [0.0, 1.0])
+assert(list(RangeFloatIterator(1, 6)) == [1.0, 2.0, 3.0, 4.0, 5.0])
+assert(list(RangeFloatIterator(0, 0)) == [])
+assert(list(RangeFloatIterator(100, 0)) == [])
+
```
Comment (dbradul) [inline Iterator.py:33]: Doesn't pass:
```
assert(list(RangeFloatIterator(10, 2, -2)) == [10, 8, 6, 4])
```

**Review by dbradul — CHANGES_REQUESTED** [review]

**Review by dbradul — APPROVED** [review]
